core/src/Cuda/Kokkos_Cuda_Parallel_Team.hpp

```diff
@@ -106,7 +106,12 @@ class TeamPolicyInternal<Kokkos::Cuda, Properties...>
             (size_t)impl_vector_length(),
             (size_t)team_scratch_size(0) + 2 * sizeof(double),
             (size_t)thread_scratch_size(0) + sizeof(double));
-    return block_size / impl_vector_length();
+    int max_block_size = block_size / impl_vector_length();
+    if (max_block_size <= 0)
```

Comment:
How could this ever be negative ?!

Reply:
It's an `int` so testing for `<=0` seems to be more appropriate than testing for `==0`.